src_c/transform.c

```diff
@@ -589,39 +607,34 @@ surf_scale(PyObject *self, PyObject *args, PyObject *kwargs)
 static PyObject *
 surf_scale_by(PyObject *self, PyObject *args, PyObject *kwargs)
 {
-    PyObject *surfobj;
-    PyObject *surfobj2 = NULL;
-    PyObject *new_args = NULL;
+    pgSurfaceObject *surfobj;
+    pgSurfaceObject *surfobj2 = NULL;
     PyObject *factorobj = NULL;
-    float scale, scaley;
-    SDL_Surface *surf;
-    int width, height;
+    float scalex, scaley;
+    SDL_Surface *surf, *newsurf;
     static char *keywords[] = {"surface", "factor", "dest_surface", NULL};
 
-    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "OO|O", keywords, &surfobj,
-                                     &factorobj, &surfobj2))
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O!O|O!", keywords,
+                                     &pgSurface_Type, &surfobj, &factorobj,
+                                     &pgSurface_Type, &surfobj2))
         return NULL;
 
-    if (!_get_factor(factorobj, &scale, &scaley)) {
+    if (!_get_factor(factorobj, &scalex, &scaley)) {
         return NULL;
     }
 
     surf = pgSurface_AsSurface(surfobj);
 
-    width = (int)(surf->w * scale);
-    height = (int)(surf->h * scaley);
-
-    if (width < 0 || height < 0)
-        return RAISE(PyExc_ValueError, "Cannot scale to negative size");
-
-    if (surfobj2)
-        new_args = Py_BuildValue("O(ii)O", surfobj, width, height, surfobj2);
-    else
-        new_args = Py_BuildValue("O(ii)", surfobj, width, height);
-    if (new_args == NULL)
+    if (!(newsurf = scale_to(surfobj, surfobj2, (int)(surf->w * scalex),
+                             (int)(surf->h * scaley))))
```

Comment:
Same here about the inline assignment, and I'd also like curly braces around this condition.